Sources/ServerSyncJob.swift

```diff
@@ -0,0 +1,146 @@
+import Foundation
+
+enum ServerSyncJob {
+    case StartJob(instanceId: String, token: String)
+    case RefreshTokenJob(newToken: String)
+    case SubscribeJob(interest: String, localInterestsChanged: Bool)
+    case UnsubscribeJob(interest: String, localInterestsChanged: Bool)
+    case SetSubscriptions(interests: [String], localInterestsChanged: Bool)
+    case ApplicationStartJob(metadata: Metadata)
+    case SetUserIdJob(userId: String)
+    case ReportEventJob(eventType: ReportEventType)
+    case StopJob
+}
+
+extension ServerSyncJob: Codable {
+    enum Discriminator: Int, Codable {
+        case StartJobKey = 0
```

Comment:
Don't only assign the first value. All cases should have `= <value>` explicitly set.

If in the future, someone added a new case in the middle (and not at the end), it would break everything!